### DISCUSSION TITLE
Problem with taking arguments with "insensitive" mode outside of query function in resolvers

### DISCUSSION THREAD

**Discussion by elarens** [prisma/prisma#11787] [Q&A]
I have a problem with taking some of the arguments outside of my query function in resolvers. I’m using Prisma Client 3.8.1 with TypeScript. Any advice would be much appreciated!

In my resolvers I have a query with a similar structure as below, and it works fine:
```
getResults: async (
    _parent: any,
    args: { params: SearchParams },
    context: Context
  ) => {
    const res = context.prisma.post.findMany({
        where: title: { contains: args.params.title, mode: "insensitive" },
    })
    return res;
  },
```
But when I try to do as below (i.e. define “where” args outside of the query), I’m getting an error (also shown below). It only occurs with _mode: "insensitive"_ flag, all other parameters are working fine, and if I remove _mode: "insensitive"_, the error disappears. But the flag is important for proper search results.

```
const getArgs = (args: { params: SearchParams }) => {
  return {
      title: { contains: args.params.title, mode: "insensitive" } 
  };
};
getResults: async (
    _parent: any,
    args: { params: SearchParams },
    context: Context
  ) => {
    const res = context.prisma.post.findMany({
        where: getArgs(args),
    })
    return res;
  },
```
Error:
```
...
Type '{ contains: string; mode: string; }' is not assignable to type 'string | StringNullableFilter | null | undefined'.
     Type '{ contains: string; mode: string; }' is not assignable to type 'StringNullableFilter'.
         Types of property 'mode' are incompatible.
             Type 'string' is not assignable to type 'QueryMode | undefined'.ts(2322)
```
P.S. I'm processing around 20 arguments in multiple queries, so this functionality is important to me.

**Comment by VajiheHabibi** [comment]
add where condition in {} 
 like this

`getResults: async (
    _parent: any,
    args: { params: SearchParams },
    context: Context
  ) => {
    const res = context.prisma.post.findMany({
        where:{ title: { contains: args.params.title, mode: "insensitive" },
        },
    })
    return res;
  },`

**Comment by piotrkulpinski** [comment]
For anyone running into similar issue with TypeScript, you have to use "insensitive" as a property of Prisma.QueryMode, like this:

```tsx
import { Prisma } from '@prisma/client'

await prisma.post.findFirst({
  where: { title: { contains: 'query', mode: Prisma.QueryMode.insensitive } },
}),
```

**Reply by azkifairuz** [reply]
hi, i have some issue with query, i dont know why my prisma dont detect mode, when i use code similar like you thi error returned
Unknown argument `mode`

this my code 

```
await prisma.community.findMany({
  where: {  community_name: { contains: query,mode:"insensitive" } },
}),
 
```

**Comment by anabrd** [comment]
Hi, I have encountered the same issue in a project running on Typescript and Prisma client 4.4.0 when trying to extract filters for a query.

Suggestion provided by @piotrkulpinski works, but I was wondering why this occurs in the first place or if anyone's found a more straightforward solution.

**Comment by gerkim62** [comment]
You can also add as const infront of the "insensitive" string like so ```"insensitive" as const```.  or use ```Prisma.QueryMode.insensitive``` as suggested by @piotrkulpinski above

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
